paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/SavedPaymentMethodRowButton.kt

```diff
@@ -35,6 +35,9 @@ internal fun SavedPaymentMethodRowButton(
         displayableSavedPaymentMethod.paymentMethod.getLabel()
             ?: displayableSavedPaymentMethod.displayName
 
+    val paymentMethodId = displayableSavedPaymentMethod.paymentMethod.id
+    val displayBrand = displayableSavedPaymentMethod.paymentMethod.card?.displayBrand
+    val testTagMetadata = if (displayBrand != null) "_$displayBrand" else ""
```

Comment:
what's the advantage of including this in the test tag instead of just accessing the PM directly in a test and ensuring it has the right brand? This seems more complex than is necessary to me

Reply:
Or if the goal is to test the icon, should we be checking that a specific icon is visible?

Reply:
I don't think we want to get access to it in the tests, then we'd have to know about a lot more implementation details. (Maybe get the activity, then get the attached view model, then get the customer state holder, etc, etc. And even then, the UI might not have updated).

My ideal here is to know the icon changed. But we already have a different test tag on the icon, at a different layer. So that threw me off for a bit, and I ended up going this route.

There's probably another way to do it. But I'm not seeing the obvious path. So if you see it, let me know!

Reply:
could we do something like setting a test tag on [this icon](https://github.com/stripe/stripe-android/blob/efc2b13d3877c2ad620e567aecd2ff2219cdd8cd/paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/SavedPaymentMethodRowButton.kt#L43-L48) equal to the resource ID of the icon being displayed? And then in the test we can verify that e.g. there's a compose node with a test tag equal to e.g. `com.stripe.payments.model.R.drawable.stripe_ic_visa_unpadded` if we expect the card brand to be visa.

Is that too implementation detail-y? 

I think this would work because we shouldn't be displaying any other card brand icons on the main vertical mode screen (since we just show one saved PM at a time)

Reply:
Oh, I see what you mean about the other layer. Because we have `TEST_TAG_ICON_FROM_RES` set on that icon. :sadpanda:

Reply:
I can shoot for a refactor that attempts that. 

This is more or less the direction I initially attempted, but was roadblocked due to https://github.com/stripe/stripe-android/blob/4c5782d803c631766a1e449997b40e6bb59b5f46/paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentMethodIcon.kt#L87

We set the test tag there, and use it elsewhere. I'll see what I can do though.